src/test/scala/ProjectTests.scala

```diff
@@ -0,0 +1,36 @@
+import org.scalatest._
+import initial._
+import scala.concurrent.duration._
+
+
+class ProjectTests extends FunSuite with DiagrammedAssertions {
+  test("Empty retry list") {
+    val expected = 2
+    val result = retry[Int](
+      block = () => 1+1,
+      acceptResult = res => res%2 == 0,
+      retries = List()
+    )
+    assert(expected == result)
+  }
+
+  test("Really big retry list") {
+    val expected = 2
+    val result = retry[Int](
+      block = () => 1+1,
+      acceptResult = res => false,
+      retries = List.tabulate(1000000)(n => 0.seconds)
+    )
+    assert(expected == result)
+  }
+
+  test("Interrupt after first attempt") {
+    val expected = 2
+    val result = retry[Int](
+      block = () => 1+1,
+      acceptResult = res => res%2 == 0,
+      retries = List(0.seconds, 1.seconds, 2.seconds)
+    )
+    assert(expected == result)
+  }
+}
```

Comment:
1. there is no assertion on how many retries the retry function does
2. there is no assertion if the retry function waits between retry attempts.